opl/cluster_read.py

```diff
@@ -373,6 +374,50 @@ def measure(self, ri, name, command, output="text"):
         return name, result
 
 
+class CountLinePlugin(BasePlugin):
+    def check_word_presence(self, word, line):
+        pattern = r"\b{}\b".format(re.escape(word))
+        return bool(re.search(pattern, line))
+
+    def measure(
+        self,
+        ri,
+        name,
+        log_source_command,
+        log_regexp_error,
+        log_regexp_warning,
+        output="text",
+    ):
+        """
+        Execute command "command" and return result as per its "output" configuration
+        """
+        result = execute(log_source_command)
+
+        regex_lst = ["log_regexp_error"[len("log_regexp_") :]]
```

Comment:
Note there can be any number of regexp. Imagine you want to process config with all of these:
```
log_regexp_error: ERROR
log_regexp_warning: WARNING
log_regexp_info: INFO
log_regexp_debug: DEBUG
```

Reply:
So I'm you need a dict (key would be regexp group name and value would be regexp) like this:
```
{
  "error": "ERROR",
  "warning": "WARNING",
  "info": "INFO",
  "debug": "DEBUG",
}
```

Reply:
The issue is not number of queries or dict produced in the end, it is the way the function is called from line no. 555. It calls it with exact number of regexp attributes and with same name.

if following example is the query:

Example:
```
log_regexp_error: '"level":"error".*"logger":"ABC"'
log_regexp_warning: '"level":"error".*"logger":"XYZ"'
```

the above query will produce the function definition with the following attributes:

```
def measure(self,
        ri,
        name,
        log_source_command,
        log_regexp_error,
        log_regexp_warning,
        output="text",
    )
```

it can neither increase or decrease the number of variable nor change their names.

To use this we will have to use a distinct number of regexp queries on call and with the same name.. we cannot change them according to each query.

Reply:
What I am thinking we can do is we can create a single query variable like log_regexp  and in this we can give a list of values to match like:
`log_regexp: ["error", "warning", "abc_error", "all_error"]`

Reply:
I was thinking about something like this code:

```
import subprocess
import re

config = {
    "name": "aaa/bbb",
    "log_source_command": 'echo -e \'{"level":"error","logger":"ABC","msg":"Oh no"}\n{"level":"info","logger":"XYZ","msg":"Hello!"}\n{"level":"error","logger":"XYZ","msg":"Oh no"}\n{"level":"warning","logger":"XYZ","msg":"Beware"}\'',
    "log_regexp_error_abc": '"level":"error".*"logger":"ABC"',
    "log_regexp_error_xyz": '"level":"error".*"logger":"XYZ"',
    "log_regexp_warning": '"level":"warning"',
}

session = subprocess.Popen(config["log_source_command"], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
stdout, stderr = session.communicate()

#print(f"STDOUT: {stdout.decode()}")
#print(f"STDERR: {stderr.decode()}")

output = {}
output["all"] = len(stdout.decode().splitlines())

for pattern_name, pattern_value in config.items():
    if not pattern_name.startswith("log_regexp_"):
        continue
    pattern_key = pattern_name[len("log_regexp_"):]
    pattern_regexp = re.compile(pattern_value)
    counter = 0
    for line in stdout.decode().splitlines():
        if pattern_regexp.search(line):
            counter += 1
    output[pattern_key] = counter

print(f"OUTPUT: {output}")
```

Output here is:

```
OUTPUT: {'all': 4, 'error_abc': 1, 'error_xyz': 1, 'warning': 1}
```